server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndex.java

```diff
@@ -371,6 +365,12 @@ private static void addDefaultFacets(RuleQuery query, SearchOptions options, Map
         stickyFacetBuilder.buildStickyFacet(FIELD_RULE_ALL_TAGS, FACET_TAGS,
           tags == null ? new String[0] : tags.toArray()));
     }
+    if (options.getFacets().contains(FACET_TYPES)) {
+      Collection<RuleType> types = query.getTypes();
+      aggregations.put(FACET_TYPES,
+        stickyFacetBuilder.buildStickyFacet(FIELD_RULE_TYPE, FACET_TYPES,
+          types == null ? new String[0] : types.toArray()));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS864)